### PR TITLE
bugfix: BB-247 use a separate set of producers for lifecycle ext

### PR DESCRIPTION
**Pull request by miniscruff**

### CODE REVIEW

File: extensions/lifecycle/LifecycleQueuePopulator.js
Comment (naren-rajendran) [inline extensions/lifecycle/LifecycleQueuePopulator.js]: Is this the expected property? I don't see `coldStorageRequest` in config.json!

File: lib/queuePopulator/QueuePopulator.js
Comment (rachedbenmustapha) [inline lib/queuePopulator/QueuePopulator.js]: Typo here

File: extensions/lifecycle/LifecycleQueuePopulator.js
Comment (rachedbenmustapha) [inline extensions/lifecycle/LifecycleQueuePopulator.js:243]: Why the `encodeURIComponent`? The rest of the codebase doesn't seem to be escaping the bucket/object key.

**Review by rachedbenmustapha — COMMENTED** [review]
I see several unrelated whitespace changes

**Comment by miniscruff** [comment]
> I see several unrelated whitespace changes

I can undo them but I seemed to have enabled auto format on save and didn't realize it until I made the PR...

File: extensions/lifecycle/LifecycleQueuePopulator.js
Comment (miniscruff) [reply extensions/lifecycle/LifecycleQueuePopulator.js:243]: I grabbed it from here: https://github.com/scality/backbeat/blob/development/8.4/lib/queuePopulator/QueuePopulatorExtension.js#L83, not sure it is required but since I was doing a publish manually I thought I would copy publish in this way.

File: extensions/lifecycle/LifecycleQueuePopulator.js
Comment (nicolas2bert) [inline extensions/lifecycle/LifecycleQueuePopulator.js:20]: `this._authConfig` is set up incorrectly, should be `params.config.conductor.auth;`

File: extensions/lifecycle/LifecycleQueuePopulator.js
Comment (nicolas2bert) [inline extensions/lifecycle/LifecycleQueuePopulator.js:224]: We need to return on error.
We should log the error.

**Review by nicolas2bert — DISMISSED** [review]

File: extensions/lifecycle/LifecycleQueuePopulator.js
Comment (nicolas2bert) [inline extensions/lifecycle/LifecycleQueuePopulator.js]: Log error if "error publishing entry".

File: extensions/lifecycle/LifecycleQueuePopulator.js
Comment (alexanderchan-scality) [reply extensions/lifecycle/LifecycleQueuePopulator.js:20]: it should be using the `auth` config defined in `config.queuePopulator`.
* https://github.com/scality/backbeat/blob/e702e5e1e1864f8bf64d7b751c995417fb9172d6/lib/config.joi.js#L34
* https://github.com/scality/backbeat/blob/e702e5e1e1864f8bf64d7b751c995417fb9172d6/lib/queuePopulator/QueuePopulator.js#L391

**Comment by miniscruff** [comment]
End to end testing was blocked by this error in the sorbet forwarder:
```
00:27:46.277979 restore http://dev-connector-s3api.default.svc.cluster.local/buck/obj04: could not assume role {{buck obj04  291129326182} {5dfdde44-583d-43d1-b6c9-be5b0f32a903 6129484611666145} f5a05065-628b-4bfc-9ae9-0a6e8c969f9d}
```
ZKOP-171 and SOR-34 were created to unblock this issue cc @alexanderchan-scality

**Review by rachedbenmustapha — APPROVED** [review]

**Review by alexanderchan-scality — APPROVED** [review]

**Review by naren-rajendran — APPROVED** [review]

File: extensions/lifecycle/LifecycleQueuePopulator.js
Comment (nicolas2bert) [inline extensions/lifecycle/LifecycleQueuePopulator.js]: Is it not the wrong log error message?

Comment (alexanderchan-scality) [reply extensions/lifecycle/LifecycleQueuePopulator.js]: @nicolas2bert @RachedBenMustapha @rahulreddy 
should a ticket be created for this to be fixed in a different PR? these changes will unblock a few other PRs

Comment (rachedbenmustapha) [reply extensions/lifecycle/LifecycleQueuePopulator.js]: @alexanderchan-scality I think we can push a change and merge this one.

Comment (alexanderchan-scality) [reply extensions/lifecycle/LifecycleQueuePopulator.js]: @RachedBenMustapha made the change and update the PR.

**Comment by rachedbenmustapha** [comment]
ping @bert-e

**Pull request merged**
